src/openforms/submissions/attachments.py

```diff
@@ -141,30 +141,22 @@ def validate_uploads(submission_step: SubmissionStep, data: Optional[dict]) -> N
         else:
             allowed_mime_types = glom(component, "file.type", default=[])
 
+        validate_file = mime_type_validator(allowed_mime_types)
+
         # perform content type validation
         with upload.content.open("rb") as infile:
-            # 2048 bytes per recommendation of python-magic
-            file_mime_type = magic.from_buffer(infile.read(2048), mime=True)
-
-        invalid_file_type_error = ErrorDetail(
-            _("The file '{filename}' is not a valid file type.").format(
-                filename=upload.file_name
-            ),
-            code="invalid",
-        )
-
-        if upload.content_type != file_mime_type:
-            validation_errors[key].append(invalid_file_type_error)
-            continue
-
-        # validate that the mime type passes the allowlist
-        if not mimetype_allowed(file_mime_type, allowed_mime_types):
-            logger.warning(
-                "Blocking submission upload %d because of invalid mime type check",
-                upload.id,
+            # wrap in UploadedFile just to reuse DRF validator
+            uploaded_file = UploadedFile(
+                file=infile, name=upload.file_name, content_type=upload.content_type
             )
-            validation_errors[key].append(invalid_file_type_error)
-            continue
+            try:
+                validate_file(uploaded_file)
+            except ValidationError as e:
+                logger.warning(
+                    "Blocking submission upload %d because of invalid mime type check",
+                    upload.id,
+                )
+                validation_errors[key].append(e.detail)
```

Comment:
do we have a check that the validation error code is  still `"invalid"` here - that information is not passed explicitly anymore in the validator (it may be the default code).